watertap/examples/flowsheets/case_studies/wastewater_resource_recovery/dye_desalination/dye_desalination.py

```diff
@@ -228,71 +442,341 @@ def set_operating_conditions(m):
+        propagate_state(prtrt.s01)
+    else:
+        pass
+
+    # initialize nf
+    seq.run(dye_sep, lambda u: u.initialize())
+
+    if hasattr(m.fs, "dewater"):
+        seq.run(m.fs.dewaterer, lambda u: u.initialize())
+        propagate_state(m.fs.s01)
+    elif hasattr(m.fs, "gac"):
+        m.fs.tb_nf_gac.properties_out[0].flow_mass_phase_comp["Liq", "H2O"] = value(
+            m.fs.tb_nf_gac.properties_in[0].flow_mass_comp["H2O"]
+        )
+        m.fs.tb_nf_gac.properties_out[0].flow_mass_phase_comp["Liq", "tds"] = value(
+            m.fs.tb_nf_gac.properties_in[0].flow_mass_comp["tds"]
+        )
+        m.fs.tb_nf_gac.properties_out[0].flow_mass_phase_comp["Liq", "dye"] = value(
+            m.fs.tb_nf_gac.properties_in[0].flow_mass_comp["dye"]
+        )
+
+        propagate_state(m.fs.s01)
+        m.fs.gac.initialize()
+    else:
+        pass
+
+    # initialize ro
+    if hasattr(m.fs, "desalination"):
+        desal = m.fs.desalination
+        propagate_state(m.fs.s_nf)
+        propagate_state(m.fs.s_ro)
+        propagate_state(desal.s01)
+        propagate_state(m.fs.s_disposal)
+        propagate_state(m.fs.s_permeate)
+
+        m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "H2O"] = value(
+            m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["H2O"]
+        )
+        m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "TDS"] = value(
+            m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["tds"]
+        ) + value(m.fs.tb_nf_ro.properties_in[0].flow_mass_comp["dye"])
+
+        desal.RO.inlet.flow_mass_phase_comp[0, "Liq", "H2O"] = value(
+            m.fs.tb_nf_ro.properties_out[0].flow_mass_phase_comp["Liq", "H2O"]
+        )
+        desal.RO.inlet.temperature[0] = value(
+            m.fs.tb_nf_ro.properties_out[0].temperature
+        )
+        desal.RO.inlet.pressure[0] = value(
+            desal.P2.control_volume.properties_out[0].pressure
+        )
+        desal.RO.initialize()
+        solve(desal, checkpoint="solve flowsheet after initializing desalination")
+    else:
+        pass
+
+    return
+
+
+def optimize_operation(m):
+    """
+    Unfixes RO operating conditions and sets solver objective
+        - Operating pressure: 1 - 83 bar
+        - Crossflow velocity: 10 - 30 cm/s
+        - Membrane area: 50 - 5000 m2
+        - Volumetric recovery: 10 - 75 %
+    """
+    desal = m.fs.desalination
+
+    # RO operating pressure
+    desal.P2.control_volume.properties_out[0].pressure.unfix()
+    desal.P2.control_volume.properties_out[0].pressure.setub(
+        8300000
+    )  # pressure vessel burst pressure
+    desal.P2.control_volume.properties_out[0].pressure.setlb(100000)
+
+    # RO inlet velocity
+    desal.RO.feed_side.velocity[0, 0].unfix()
+    desal.RO.feed_side.velocity[0, 0].setub(0.3)
+    desal.RO.feed_side.velocity[0, 0].setlb(0.1)
+
+    # RO membrane area
+    desal.RO.area.unfix()
+    desal.RO.area.setub(5000)
+    desal.RO.area.setlb(50)
```

Comment:
I wonder if we want to check these values in tests since they could play a role in convergence later on if we make changes.

Reply:
I will be sure to include checks for these in cases where RO is in the flowsheet